File: app/src/main/java/com/example/kineduexample/ui/main/MainPresenter.java
package com.example.kineduexample.ui.main;

import com.example.kineduexample.data.network.KineduInteractor;

public class MainPresenter {
    private KineduInteractor interactor;
    private MainView view;

    public MainPresenter(KineduInteractor interactor){
        this.interactor = interactor;
    }

    public void bind(MainView view){
        this.view = view;
    }

    public void unbind(){
        view = null;
    }


    public Integer getSelectedSpinnerValue(String value) {
        try{
            if(value.contains("0-1 MONTH")){
                return 1;
            }
            return Integer.valueOf(value.replace("MONTH", "").replace("S", "").replace("ALL","").trim());
        }catch (Exception e){
            e.printStackTrace();
            return -1;
        }
    }
}

File: app/src/main/java/com/example/kineduexample/ui/main/MainPresenter.kt
package com.example.kineduexample.ui.main

import com.example.kineduexample.data.network.KineduInteractor

class MainPresenter(private val interactor: KineduInteractor) {
    private var view: MainView? = null

    fun bind(view: MainView) {
        this.view = view
    }

    fun unbind() {
        view = null
    }


    fun getSelectedSpinnerValue(value: String): Int? {
        try {
            return if (value.contains("0-1 MONTH")) {
                1
            } else Integer.valueOf(value.replace("MONTH", "").replace("S", "").replace("ALL", "").trim { it <= ' ' })
        } catch (e: Exception) {
            e.printStackTrace()
            return -1
        }

    }
}
